test/simplify_reshapes_test.cpp

```diff
@@ -1077,6 +1077,53 @@ TEST_CASE(multibroadcast_simplify)
     EXPECT(std::distance(m.begin(), m.end()) == n - 1);
 }
 
+TEST_CASE(multibroadcast_unsqueeze_scalar)
```

Comment:
minor: can we add a note or something to make it more obvious which matcher this is testing?